src/dialect/mod.rs

```diff
@@ -40,4 +40,37 @@ pub trait Dialect: Debug {
     fn is_identifier_start(&self, ch: char) -> bool;
     /// Determine if a character is a valid unquoted identifier character
     fn is_identifier_part(&self, ch: char) -> bool;
+
+    /// The name of the dialect
+    fn dialect_name(&self) -> &'static str;
```

Comment:
I tried to take the same approch

Reply:
Since GitHub thinks this branch of discussion is "outdated', I continued [below](https://github.com/ballista-compute/sqlparser-rs/pull/254#issuecomment-668044027).